lib/katello/tasks/upgrades/3.0/update_subscription_facet_backend_data.rake

```diff
@@ -1,10 +1,40 @@
 namespace :katello do
   namespace :upgrades do
     namespace '3.0' do
-      task :update_subscription_facet_backend_data => ["environment"] do
+      task :update_subscription_facet_backend_data => ["environment"]  do
+
+        def error(message)
+          @errors << message
+          @errors << "\n"
+        end
+
+        def report_errors
+          if @errors.any?
+            file = File.open("/tmp/upgrade-#{Time.now.to_i}", 'w')
```

Comment:
Will foreman-debug pick up the /tmp/upgrade-* file?  If not, would it be worthwhile to place the file under /var/log/foreman-installer?

Since it is a temporary file, that may not be ideal; however, it would ensure that we get access to the information if/when users report errors.